functionary/ui/forms/tasks.py

```diff
@@ -160,3 +165,76 @@ def _get_field_info(
         the field class and widget based on the input data.
         """
         return _field_mapping.get(parameter_type, (None, None))
+
+
+class TaskParameterTemplateForm(TaskParameterForm):
+    """TaskParameterForm variant with template variable support
+
+    This is a version of TaskParameterForm that allows template variables to be entered
+    as values regardless of parameter type. That is, a field requiring an integer will
+    allow either and integer value or a template variable such as {{step.result}}.
+    Other than the exception for template variables, other validation rules still apply.
+    """
+
+    def _stringify_template_variables(self, parameter_template: str) -> str:
+        """Converts the template variables in the parameter template to strings so
+        that the template can be safely jsonified.
+        """
+        return re.sub(r"{{([\w\.]+)}}", r'"{{\1}}"', parameter_template)
+
+    def _build_parameter_template(self, parameters: dict) -> str:
+        """Undo the template variable stringification that was required to jsonify
+        the template. The resulting string is one that can be rendered as a django
+        template"""
+        json_data = json.dumps(parameters)
+
+        return re.sub(r'"{{([\w\.]*)}}"', r"{{\1}}", json_data)
+
+    def __init__(
+        self,
+        function: "Function",
+        data: Optional["QueryDict"] = None,
+        initial: str | None = None,
+    ):
+        """TaskParameterTemplateForm init
+
+        Args:
+            function: The Function that will be executed when the WorkflowStep runs
+            data: Form submission data
+            initial: Initial values to populate the form fields with on render
+        """
+        initial_data = (
+            json.loads(self._stringify_template_variables(initial)) if initial else {}
+        )
+
+        super().__init__(function=function, data=data, initial=initial_data)
```

Comment:
I got the following error when trying to edit a step that I set the `load_json()` parameter to `{{param1}}`:
`json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)`

Reply:
I fixed this in a way that I think I can live with for now.  Basically, if the json loads fails for the initial values, I just have it return the original string value.  The result is a little bit weird because when you go to edit a step with a json parameter containing a template variable, the variable now shows up as quoted.  

![image](https://user-images.githubusercontent.com/86254668/212979562-54175eef-a71f-4500-8f6e-8203d2d2e3d8.png)

The behavior ends up being correct, whether you leave the quotes in or take them out, so I think I can live with that for now.

I also added a unit test for the scenario that was failing, which now passes with the change.

Reply:
That works for me.